exchanges/poloniex/poloniex_wrapper.go

```diff
@@ -491,66 +667,189 @@ func (e *Exchange) SubmitOrder(ctx context.Context, s *order.Submit) (*order.Sub
 	if err := s.Validate(e.GetTradingRequirements()); err != nil {
 		return nil, err
 	}
-
-	fPair, err := e.FormatExchangeCurrency(s.Pair, s.AssetType)
+	var err error
+	s.Pair, err = e.FormatExchangeCurrency(s.Pair, s.AssetType)
 	if err != nil {
 		return nil, err
 	}
-	response, err := e.PlaceOrder(ctx,
-		fPair.String(),
-		s.Price,
-		s.Amount,
-		false,
-		s.Type == order.Market,
-		s.Side.IsLong())
-	if err != nil {
-		return nil, err
+	switch s.AssetType {
+	case asset.Spot:
+		var stpMode string
+		switch s.TimeInForce {
+		case order.PostOnly:
+			stpMode = "EXPIRE_MAKER"
+		case order.GoodTillCancel:
+			stpMode = "EXPIRE_TAKER"
+		}
+		switch s.Type {
+		case order.Stop, order.StopLimit, order.TrailingStop, order.TrailingStopLimit:
+			var trackingDistance string
+			if s.Type|order.TrailingStop == order.TrailingStop {
+				if s.TrackingMode == order.Percentage {
+					trackingDistance = fmt.Sprintf("%f%%", s.TrackingValue)
+				} else {
+					trackingDistance = strconv.FormatFloat(s.TrackingValue, 'f', -1, 64)
+				}
```

Comment:
🚧 Change request:
These same like the same thing, but using different methods for the same thing.
Can we use the same method and then just suffix, please?
```suggestion
				trackingDistance = strconv.FormatFloat(s.TrackingValue, 'f', -1, 64)
				if s.TrackingMode == order.Percentage {
					trackingDistance += "%"
				}
```